src/modules/headset/headset_openxr.c

```diff
@@ -1773,6 +1823,63 @@ static bool openxr_animate(Device device, Model* model) {
   return true;
 }
 
+static bool openxr_animateMSFT(XrControllerModelKeyMSFT modelKey, Model* model) {
+  XrControllerModelNodePropertiesMSFT nodeProperties[16];
```

Comment:
OpenXR might make us fill in `type` for each element of `nodeProperties` and `nodeStats`, sort of like `extensionProperties` is doing.  Not 100% sure on it though.

Reply:
yeah, I was unsure about this but did to me the spec reads like initialization is up to the runtime here?

> If propertyCapacityInput is not 0, properties must be a pointer to an array of propertyCapacityInput [XrExtensionProperties](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XrExtensionProperties) structures

vs

> nodeProperties is a pointer to an application-allocated array that will be filled with the [XrControllerModelNodePropertiesMSFT](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XrControllerModelNodePropertiesMSFT) values. It can be NULL if nodeCapacityInput is 0.

Reply:
I just carefully read the section on [buffer size parameters](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#buffer-size-parameters) and you're right actually:

> In case that elements is a struct with type and next fields, the application must set the type to the correct value as well as next either to NULL or a struct with extension related data in which type and next also need to be well defined.